tests/utils/vector.c

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright 2019 UChicago Argonne, LLC.
+ * (c.f. AUTHORS, LICENSE)
+ *
+ * This file is part of the libnrm project.
+ * For more info, see https://github.com/anlsys/libnrm
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ ******************************************************************************/
+
+#include "nrm.h"
+#include <check.h>
+#include <stdlib.h>
+
+START_TEST(test_basics)
+{
+	nrm_vector_t *vector = NULL;
+	nrm_vector_create(&vector, sizeof(nrm_string_t));
+	int a = 42;
+	int err;
+	nrm_string_t fortytwo = nrm_string_fromchar("fortytwo");
+	ck_assert_str_eq("fortytwo", fortytwo);
+
+	err = nrm_vector_push_back(vector, &fortytwo);
+	ck_assert_int_eq(err, 0);
+	ck_assert_ptr_nonnull(vector);
+
+	size_t pos;
+	err = nrm_vector_find(vector, &fortytwo, nrm_string_cmp, &pos);
+	ck_assert_int_eq(err, 0);
+
+	size_t len;
+	err = nrm_vector_length(vector, &len);
+	ck_assert_int_eq(err, 0);
+	ck_assert_int_eq(len, 1);
+
+	nrm_vector_destroy(&vector);
+	ck_assert_ptr_null(vector);
+	nrm_string_decref(fortytwo);
+}
+
+START_TEST(test_iter)
+{
+	nrm_vector_t *vector = NULL;
+	nrm_vector_create(&vector, sizeof(nrm_string_t));
+	int a = 42, b = 43, c = 44;
+	int err;
+	nrm_string_t fortytwo = nrm_string_fromchar("fortytwo");
+	nrm_string_t fortythree = nrm_string_fromchar("fortythree");
+	nrm_string_t fortyfour = nrm_string_fromchar("fortyfour");
+	ck_assert_str_eq("fortytwo", fortytwo);
+	err = nrm_vector_push_back(vector, &fortytwo);
+	ck_assert_int_eq(err, 0);
+	err = nrm_vector_push_back(vector, &fortythree);
+	ck_assert_int_eq(err, 0);
+	err = nrm_vector_push_back(vector, &fortyfour);
+	ck_assert_int_eq(err, 0);
+
+	nrm_string_t str;
+	nrm_vector_foreach(vector, iterator)
+	{
+		str = nrm_vector_iterator_get(iterator);
+	}
```

Comment:
you should be able to write a test in there that confirms the value of each element.
The easiest would be to make an array of the values pushed in the vector, and check each of them in the loop